crates/networking/discv5/src/discovery.rs

```diff
@@ -51,65 +56,145 @@ struct QueryResult {
 
 pub struct Discovery {
     discv5: Discv5,
+    local_enr: Enr,
     event_stream: EventStream,
     discovery_queries: FuturesUnordered<Pin<Box<dyn Future<Output = QueryResult> + Send>>>,
     find_peer_active: bool,
     pub started: bool,
+    subnets: Subnets,
 }
 
 impl Discovery {
-    pub async fn new(local_key: Keypair, config: &NetworkConfig) -> anyhow::Result<Self> {
-        let enr_local = convert_to_enr(local_key)?;
-        let enr = Enr::builder().build(&enr_local).unwrap();
+    pub async fn new(
+        local_key: libp2p::identity::Keypair,
+        config: &NetworkConfig,
+    ) -> anyhow::Result<Self> {
+        let enr_key = convert_to_enr(local_key)
+            .map_err(|e| anyhow::anyhow!("Failed to convert key: {:?}", e))?;
+        let mut enr_builder = Enr::builder();
+        enr_builder.ip(config.socket_address);
+        enr_builder.udp4(config.socket_port);
+        if let Some(attestation_bytes) = config.subnets.attestation_bytes() {
+            enr_builder.add_value::<Bytes>(ATTESTATION_BITFIELD_ENR_KEY, &attestation_bytes);
+        }
+        let enr = enr_builder
+            .build(&enr_key)
+            .map_err(|e| anyhow::anyhow!("Failed to build ENR: {:?}", e))?;
         let node_local_id = enr.node_id();
 
-        let mut discv5 = Discv5::new(enr, enr_local, config.discv5_config.clone())
-            .map_err(|err| anyhow!("Failed to create discv5: {err:?}"))?;
+        let mut discv5 = Discv5::new(enr.clone(), enr_key, config.discv5_config.clone())
+            .map_err(|e| anyhow::anyhow!("Failed to create discv5: {:?}", e))?;
 
-        // adding bootnodes to discv5
-        for enr in config.bootnodes.clone() {
-            // Skip adding ourselves to the routing table if we are a bootnode
-            if enr.node_id() == node_local_id {
+        for bootnode_enr in config.bootnodes.clone() {
+            if bootnode_enr.node_id() == node_local_id {
                 continue;
             }
-
-            if let Err(err) = discv5.add_enr(enr) {
-                error!("Failed to add bootnode to Discv5 {err:?}");
-            };
+            discv5
+                .add_enr(bootnode_enr)
+                .map_err(|e| anyhow::anyhow!("Failed to add bootnode: {:?}", e))?;
```

Comment:
this should be reverted back to what it was

Reply:
done , reverted to original .